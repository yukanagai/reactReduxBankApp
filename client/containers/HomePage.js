import React from 'react';
import { Link } from 'react-router';
import ButtonWithIcon from '../components/ButtonWithIcon.js';

class HomePage extends React.Component {

	constructor(props) {
		super(props);
	}

	showModal(modalType) {
		// CALLING TOGGLE MODAL ACTION TO SWITCH MODAL VIEWS
		this.props.toggleModal(modalType);
	}

	render() {
		return (
			<div className='home'>
				<div className='content'>
					<p>ðŸ’°ðŸ’°ðŸ’°</p>
					<p>Your current balance is: <span>${this.props.balance.balance}</span></p>
				</div>
				<div className='buttonSection'>
					<Link to='/modal/DEPOSIT'>
						<ButtonWithIcon
							id='deposit'
							iconClass='plus icon'
							buttonText='deposit'
							onClickAction={this.showModal.bind(this, 'DEPOSIT')} />
					</Link>
					<Link to='/modal/WITHDRAW'>
						<ButtonWithIcon
							id='withdraw'
							iconClass='minus icon'
							buttonText='withdraw'
							onClickAction={this.showModal.bind(this, 'WITHDRAW')} />
					</Link>
				</div>
			</div>
		);
	};
}

export default HomePage;