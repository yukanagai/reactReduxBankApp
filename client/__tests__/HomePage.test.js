import React from 'react';
import renderer from 'react-test-renderer';
import { shallow, mount } from 'enzyme';
import HomePage from '../containers/HomePage.js';

// Snapshot for HomePage React Component
test('HOMEPAGE --- Snapshot',()=>{
  const tree = renderer.create(
  	<HomePage balance={3000} />
  ).toJSON();
 	expect(tree).toMatchSnapshot();
});

describe('HOMEPAGE --- Shallow Render REACT COMPONENTS',()=>{
  let wrapper;
  const balance=3000;

  beforeEach(()=>{
  	wrapper = shallow(<HomePage balance={balance}/>)
  });
  
});