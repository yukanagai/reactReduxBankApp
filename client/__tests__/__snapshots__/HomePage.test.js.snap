// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>>HOMEPAGE --- Snapshot 1`] = `
<div
  className="home"
>
  <div
    className="content"
  >
    <p>
      ðŸ’°ðŸ’°ðŸ’°
    </p>
    <p>
      Your current balance is: 
      <span>
        $
      </span>
    </p>
  </div>
  <div
    className="buttonSection"
  >
    <a
      onClick={[Function]}
      style={Object {}}
    >
      <button
        className="ui button"
        id="deposit"
        onClick={[Function]}
      >
        <i
          className="plus icon"
        />
        deposit
      </button>
    </a>
    <a
      onClick={[Function]}
      style={Object {}}
    >
      <button
        className="ui button"
        id="withdraw"
        onClick={[Function]}
      >
        <i
          className="minus icon"
        />
        withdraw
      </button>
    </a>
  </div>
</div>
`;
